src/reducers/cacheReducer.js

```diff
@@ -232,6 +241,73 @@ const filterTransducers = (where) => {
     );
   });
 };
+
+/**
+ * @name paginateTransducers
+ * @param {RRFQuery} query - Firestore wquery
+ * @typedef {Function} xFormFilter - run the same where cause sent to
+ * Only apply pagination if we get the full list of docs, pre-sorted
+ * @returns {xFormFilter} - transducer
+ */
+const paginateTransducers = (query) => {
+  const { orderBy: order, startAt, startAfter, endAt, endBefore, via } = query;
+  const isOptimisticRead = via === undefined;
+  if (!isOptimisticRead) return null;
+
+  const start = startAt || startAfter;
+  const end = endAt || endBefore;
+  const isAfter = startAfter !== undefined;
+  const isBefore = endBefore !== undefined;
+  if (start === undefined && end === undefined) return null;
+
+  const isFlat = typeof order[0] === 'string';
+  const orders = isFlat ? [order] : order;
+  const isPaginateMatched = (doc, at, before = false, after = false) => {
+    let matched = null;
+    orders.forEach(([field, sort = 'asc'], idx) => {
```

Comment:
I think that the `Array.prototype.find()` [statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) is more convenient in here. 
Isn't it?

Reply:
done.